File: GooseClient/src/sample/mina/Message.java
package sample.mina;

import java.text.SimpleDateFormat;

/**
 * @author DeMon
 * Created on 2020/5/15.
 * E-mail 757454343@qq.com
 * Desc:
 */
public class Message {
    private int type = 0;
    private String name;
    private String content;
    private long time;
    private String to;


    public Message(String content) {
        this.type = 1;
        this.content = content;
        this.name = MacHelper.getInstance().getSelfMacAddress();
        this.time = System.currentTimeMillis();
    }

    public Message(String content, String to) {
        this.type = 0;
        this.content = content;
        this.name = MacHelper.getInstance().getSelfMacAddress();
        this.time = System.currentTimeMillis();
        this.to = to;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public long getTime() {
        return time;
    }

    public void setTime(long time) {
        this.time = time;
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }

    public String getTo() {
        return to;
    }

    public void setTo(String to) {
        this.to = to;
    }

    @Override
    public String toString() {
        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
        String obj = "";
        if (!MacHelper.getInstance().getSelfMacAddress().equals(this.name)) {
            obj = "[接收]";
        }
        return "[" + sdf.format(this.time) + "]" + obj + "[" + this.name + "]：" + this.content;
    }
}

File: MinaApp/app/src/main/java/com/demon/imapp/data/Message.kt
package com.demon.imapp.data

import android.annotation.SuppressLint
import com.demon.imapp.util.SPUtil
import com.demon.imapp.util.gson
import java.text.SimpleDateFormat

/**
 * @author DeMon
 * Created on 2020/5/14.
 * E-mail 757454343@qq.com
 * Desc:
 */
class Message {
    var type: Int = 0
    var name: String = ""
    var content: String = ""
    var time: Long = 0L
    var to: String = ""

    constructor(content: Any?) {
        val msg = gson.fromJson(content.toString(), Message::class.java)
        this.type = msg.type
        this.name = msg.name
        this.content = msg.content
        this.time = msg.time
    }

    constructor(content: String) {
        this.content = content
        this.name = SPUtil.getInstance().get(Const.NAME, "") as String
        this.to = SPUtil.getInstance().get(Const.FRIEND, "") as String
        this.time = System.currentTimeMillis()
    }

    constructor(name: String, content: String) {
        this.name = name
        this.content = content
        this.time = System.currentTimeMillis()
    }

    constructor(type: Int) {
        this.type = type
        this.content = "[初始服务器注册]"
        this.name = SPUtil.getInstance().get(Const.NAME, "") as String
        this.time = System.currentTimeMillis()
        this.to = "[服务器]"
    }


    @SuppressLint("SimpleDateFormat")
    fun getTimeStr(): String {
        val sdf = SimpleDateFormat("yyyy-MM-dd HH:mm:ss")
        return sdf.format(this.time)
    }

    override fun toString(): String {
        return "[${getTimeStr()}]-$name：$content"
    }
}